app/helpers/automated_tests_helper.rb

```diff
@@ -738,9 +737,53 @@ def self.launch_test(assignment, repo_path, call_on)
     end
   end
 
-  def self.process_result(result, grouping_id, assignment_id)
-  end
+  def self.process_result(raw_result)
+    result = Hash.from_xml(raw_result)
+    repo = @grouping.group.repo
+    @revision = repo.get_latest_revision
+    @revision_number = @revision.revision_number
```

Comment:
I definitely don't think they should be instance variables but I followed
the pattern used by the previous implementation and other methods in the
helper file.

I'm more than happy to make that change. I just wasnt sure what impact it
would have on other methods if any - I guess I assumed there was a good
reason for it being instance variables previously but haven't fully
investigated.
On 20 Nov 2015 6:38 pm, "David Liu" <notifications@github.com> wrote:

> In app/helpers/automated_tests_helper.rb
> <https://github.com/MarkUsProject/Markus/pull/2362#discussion_r45540107>:
>
> > @@ -738,9 +737,53 @@ def self.launch_test(assignment, repo_path, call_on)
> >      end
> >    end
> >
> > -  def self.process_result(result, grouping_id, assignment_id)
> > -  end
> > +  def self.process_result(raw_result)
> > +    result = Hash.from_xml(raw_result)
> > +    repo = @grouping.group.repo
> > +    @revision = repo.get_latest_revision
> > +    @revision_number = @revision.revision_number
>
> I'm a little skeptical that this and revision need to be instance
> variables... have you thought about this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MarkUsProject/Markus/pull/2362/files#r45540107>.
>